server_test.go

```diff
@@ -10,6 +11,8 @@ import (
 	"sync"
 	"testing"
 	"time"
+
+	"golang.org/x/sync/errgroup"
```

Comment:
hmmm this adds a dep, just for a test. Do we really need it?

Reply:
Without doing these network operations in parallel the test runs for several minutes. That being said, I can probably throw together a simple replacement that should work without added dependencies. I’ve sketched something that’s 20 lines that should work, I’ll try it out when I get to my computer.

Reply:
It would basically end up as a reimplementation of x/sync/errgroup. I’d rather just use that.